source/common/network/cidr_range.h

```diff
@@ -117,6 +120,23 @@ class CidrRange {
   int length_;
 };
 
+/**
+ * Class for keeping a list of CidrRanges, and then determining whether an
+ * IP address is in the CidrRange list.
+ */
+class IpList {
+public:
+  IpList(const std::vector<std::string>& subnets);
+  IpList(const Json::Object& config, const std::string& member_name);
```

Comment:
I don't feel that strongly (I see that this was there before), but I tend to feel that the principle of minimalism would dictate this be a static factory, since you can express the JSON variation in terms of the existing constructor, it doesn't depend interact with other method or state in the object.